src/events/tls.py

```diff
@@ -136,8 +137,12 @@ def _trusted_relation_created(self, event: EventBase) -> None:
             event.defer()
             return
 
-        # Create a "mtls" flag so a new listener (CLIENT_SSL) is created
-        self.charm.state.cluster.update({"mtls": "enabled"})
+        if not self.charm.state.cluster.mtls_enabled:
+            # Create a "mtls" flag so a new listener (CLIENT_SSL) is created
+            self.charm.state.cluster.update({"mtls": "enabled"})
+            self.charm.on.config_changed.emit()
+
+        self.charm.app.status = ActiveStatus()
```

Comment:
**todo**: Would you mind checking if we really need this statement, please? I missed that during my first review and I saw it in the k8s PR, but we recently changed the way we handle app/unit statuses by using the `collect-status` event, which defaults to active